### ISSUE TITLE
Loss not decreasing on a new dataset

### ISSUE THREAD

**Issue by aloyspx** [Luffy03/Large-Scale-Medical#33]
Hello,

Thank you for your work on the VoCo model and for sharing the code. I'm attempting to adapt your pipeline for a head CT segmentation task, but I am encountering issues with loss convergence.

### Context & Problem

I have a dataset of ~25,000 head CT scans. I've used your chest CT transformation pipeline as a base (as the data shares similar voxel spacing characteristics), and simply modified the HU windowing to appropriate values for head CT.
   
Despite these adaptations, after 100,000+ training steps with the Voco model, the training loss is not decreasing and remains stagnant around ~1.15-1.2. The original VoCo paper used ~10k images, so data quantity shouldn't be an issue ? I've also already used this dataset with other SSL methods and have achieved good results, which leads me to believe the quality of the data is not in play.

### Questions

I'd be grateful for any insights on:

- Potential reasons for the lack of convergence. Or is this the normal behaviour?
- Specific parameters or pipeline components that might need tuning for anisotropic head CT data with VoCo.

Thank you for considering my questions.

**Comment by aloyspx** [comment]
Thank you for the quick answer !

I hope my answers can help clarify:

- Training settings: I tried to minimally change the hyperparameters defined in your training script (voco_train.py). The only hyperparameter that I have altered is the maximum number of steps, which I have set to 200,000 for an initial trial. 
- Dataset information: [head CT](https://en.wikipedia.org/wiki/Computed_tomography_of_the_head) can be referred to as brain CT, but no brain extraction was performed. The dataset has ~25,000 cases with median voxel spacing ~[0.5, 0.5, 5.0] and ~30-35 axial slices per volume. 
- Other SSL methods: I have had some good results with MAE and BYOL with different hierarchical transformer encoders like Swin.
- GPU details: a single A6000 for the current trial.

Please find the log attached for my ongoing run:

[log.txt](https://github.com/user-attachments/files/20497191/log.txt)

**Comment by Luffy03** [comment]
Thanks a lot for your valuable feedbacks. It seems that the loss is not convergent, which has not been observed in our experiments. In our work, we did not use datasets like [head CT](https://en.wikipedia.org/wiki/Computed_tomography_of_the_head). We may need to further evaluate the effectiveness of VoCo on this kind of dataset. 
Then, have you ever tested the downstream segmentation performance with our pre-trained models? And what is the downstream segmentation performance of MAE and BYOL? We would be highly grateful if you could provide some downstream results for reference.

**Comment by aloyspx** [comment]
I have not tested the effectiveness of your pretrained models on the downstream task: intracranial hemorrhage (ICH) segmentation. Of course, we can test this, but I was curious to use our unlabelled head CT dataset to pretrain with your method, as it would probably have better domain alignment with the downstream task.
At this time, I can't share anything concrete about our downstream segmentation performance. But in general, we are able to achieve an increase in performance across metrics compared to some popular methods like nnUNet and STU-Net when using pretrained encoders.

**Comment by Luffy03** [comment]
It is encouraging to hear that SSL can improve your downstream tasks, and we are looking forward to your downstream evaluation results of VoCo. We will further test the pre-training of VoCo on this kind of dataset.
Thanks a lot for your comments!
